UI/src/csvImporter.js

```diff
@@ -0,0 +1,216 @@
+    var disableStatus = _.isEmpty($('.municipalities').find("#municipalities_search_to, select[name*='municipalityNumbers']").find('option'));
+    $('.btn.btn-primary.btn-lg').prop('disabled', disableStatus);
+  });
+
+  $('#csvImport').on('submit', (function(e) {
+    e.preventDefault();
+    var formData = new FormData(this);
+    var assetType = $('#asset-selection').find(":selected").val();
+    function uploadFile() {
+      backend.uploadFile(formData, assetType,
+        function() {
+          spinnerOff();
+          getJobs();
+        },
+        function(xhr) {
+          spinnerOff();
+          if(xhr.status === 403)
+            alert("Vain operaattori voi suorittaa Excel-ajon");
+          getJobs();
+        });
+    }
+    if ($('#deleteCheckbox').is(':checked')) {
+      new GenericConfirmPopup('Haluatko varmasti poistaa kaikki jo aiemmin kunnan alueelle lisätyt liikennemerkit?', {
+        successCallback: function () {
+          var optionValues = $('.municipalities').find("#municipalities_search_to, select[name*='municipalityNumbers']").find('option');
+          _.each(optionValues, function (opt) {
+            opt.selected = true;
+          });
+          uploadFile();
+          spinnerOn();
+        }
+      });
+    } else {
+      uploadFile();
+      spinnerOn();
+    }
+  }));
+
+  function getMunicipalities() {
+    if (_.isEmpty($('.municipalities').find("#municipalities_search").find('option'))) {
+      if (_.isEmpty(municipalities)) {
+        backend.getMunicipalities(
+          function(result){
+            municipalities = result;
+            setMunicipalities();
+          },
+          function(){
+            municipalities = [];
+          }
+        );
+      } else
+        setMunicipalities();
+    }
+  }
+
+  function setMunicipalities() {
+    _.forEach(municipalities, function (municipality) {
+      $('.municipalities').find("#municipalities_search").append($('<option>', {
+        value: municipality.id,
+        text: municipality.name
+      }));
+    });
+  }
+
+  $('#municipalities_search').multiselect({
+    search: {
+      left:
+        '<label class="control-label labelBoxLeft">Kaikki kunnat</label>' +
+        '<input type="text" id = "left_municipalities" class="form-control" placeholder="Kuntanimi" />',
+      right:
+        '<label class="control-label labelBoxRight">Valitut Kunnat</label>' +
+        '<input type="text" id = "right_municipalities" class="form-control" placeholder="Kuntanimi" />'
+    },
+    fireSearch: function(value) {
+      return value.length >= 1;
+    }
+  });
+
+  function getJobs() {
+     backend.getJobs().then(function(jobs){
+      if(!_.isEmpty(jobs)) {
+        $('.job-status').empty().html(buildJobTable(jobs));
+        rootElement.find('.job-status-link').on('click', function (event) {
+          getJob(event);
+        });
+        if(!refresh)
+          refresh = setInterval(refreshJobs, 30000);
```

Comment:
what should be the interval for fetching new jobs?

Reply:
I would say something like 5 to 10 seconds. Seems enough to me.